src/vs/platform/update/electron-main/updateService.snap.ts

```diff
@@ -191,20 +191,16 @@ export class SnapUpdateService extends AbstractUpdateService2 {
 	protected doQuitAndInstall(): void {
 		this.logService.trace('update#quitAndInstall(): running raw#quitAndInstall()');
 
-		if (typeof process.env.SNAP === 'undefined') {
-			return;
-		}
-
 		// Allow 3 seconds for VS Code to close
-		spawn('bash', ['-c', path.join(process.env.SNAP, `usr/share/${product.applicationName}/snapUpdate.sh`)], {
+		spawn('bash', ['-c', path.join(process.env.SNAP!, `usr/share/${product.applicationName}/snapUpdate.sh`)], {
```

Comment:
It's missing in the current VS Code snap package: https://snapcraft.io/vscode This package isn't release by us, so we don't support it. The current implementation of the snap update service wasn't made for that package, but for our official one which we are currently on the way of releasing. It will contain the `snapUpdate.sh` script.

I'm not too keen on merging right now this since this will very likely break the current VS Code snap package. Is the package already broken today, and this PR fixes it? How did you realize you had missed a change?

Reply:
Thank you for your response. I am glad to hear that Microsoft is creating a Snap package. Snap packages I find are easier to maintain, distribute and are much safer than ppas and debs.

>I'm not too keen on merging right now this since this will very likely break the current VS Code snap package. Is the package already broken today, and this PR fixes it? 

This code currently doesn't work and isn't being expressed by the current Snap package by snapcrafters. I currently don't have access to a snap mounted as code. All I know is that the path '/snap/${product.applicationName}/current' is more fragile compared to `${path.dirname(process.env.SNAP!)}/current`. Both will probably work for the internal snap, but the latter will work with other snaps too. 

I was just casually looking through the codebase to find areas to work and improve. I am still trying to understand how the updateService.linux.ts and updateService.snap.ts works in general. Both seem to offer some handling for snap installs.

Reply:
Still, note that there are quite some (big) issues with snap packages: https://github.com/Microsoft/vscode/issues?q=is%3Aissue+is%3Aopen+sort%3Aupdated-desc+label%3Asnap We're also waiting on fixes for those before committing to an official snap package.